src/main/java/vn/zalopay/benchmark/GRPCSampler.java

```diff
@@ -70,37 +73,53 @@ private void initGrpcClient() {
 
     @Override
     public SampleResult sample(Entry ignored) {
-        GrpcResponse grpcResponse = new GrpcResponse();
         SampleResult sampleResult = new SampleResult();
 
-        // Console prints unknown exceptions - Intercepts unknown exceptions and sends them to the console to print instead of throwing the results into the GRPC response.
+        // Intercepts exceptions before GRPC requests are initiated
         try {
+            sampleResult.setSampleLabel(getName());
             initGrpcConfigRequest();
             initGrpcClient();
-            sampleResult.setSampleLabel(getName());
             String grpcRequest = clientCaller.buildRequestAndMetadata(getRequestJson(), getMetadata());
             sampleResult.setSamplerData(grpcRequest);
             sampleResult.setRequestHeaders(clientCaller.getMetadataString());
             sampleResult.sampleStart();
         } catch (Exception e) {
-            log.error("An unknown error occurred before the GRPC request was initiated, and the stack trace is as follows: ", e);
-            startSamplerToStopIfCatchException(sampleResult);
```

Comment:
this method should be included when catching error in initiated stage, or we will get the error related to "...SampleResult: setEndTime must be called after setStartTime"

Reply:
Hello, no need to call here `sampleResult.sampleEnd();`

Reply:
thanks @yl-yue